src/charm.py

```diff
@@ -51,16 +54,18 @@ def _on_config_changed(self, event):
         if not isinstance(self.unit.status, ActiveStatus):
             event.defer()
             return
+        broker_rack = self.config.get("broker-rack")
+        if broker_rack is None:
+            self.unit.status = BlockedStatus("broker-rack config missing, please set a value") 
 
+        content = f"broker.rack={broker_rack}"
         rack_properties_file = "/var/snap/charmed-kafka/current/etc/kafka/rack.properties"
-        broker_rack = self.config["broker-rack"]
 
-        content = f"broker.rack={broker_rack}"
         safe_write_to_file(content=content, path=rack_properties_file)
 
         # Normalize file to have same owner as kafka charm files
         shutil.chown(rack_properties_file, user="snap_daemon", group="root")
```

Comment:
**praise:** Really good pre-emptive catch on the file permissions! I hadn't thought of that case!